app/controllers/spam_domains_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class SpamDomainsController < ApplicationController
+  before_action :check_if_smokedetector, only: [:create]
+  before_action :authenticate_user!, only: [:edit, :update, :destroy]
+  before_action :verify_core, only: [:edit, :update]
+  before_action :verify_admin, only: [:destroy]
+  before_action :set_spam_domain, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @total = SpamDomain.count
+    @domains = if params[:filter].present?
+                 SpamDomain.where("domain LIKE '%#{params[:filter]}%'")
+               else
+                 SpamDomain.all
+               end.order(domain: :asc).paginate(page: params[:page], per_page: 100)
+    @counts = SpamDomain.where(id: @domains.map(&:id)).joins(:posts).group('spam_domains.id').count
+  end
+
+  def create
+    @post = Post.find params[:post_id]
+    domains = params[:domains]
+    domains.each do |d|
+      record = SpamDomain.find_or_create_by domain: d
+      @post.spam_domains << record
+    end
+
+    render json: { status: 'success', total_domains: @post.spam_domains.count }
+  end
+
+  def show
+    @counts = { all: SpamDomain.where(id: params[:id]).joins(:posts).count,
+                tp: SpamDomain.where(id: params[:id]).joins(:posts).where(posts: { is_tp: true }).count }
+    @posts = @domain.posts.includes_for_post_row
+    @sites = Site.where(id: @posts.map(&:site_id))
+  end
+
+  def edit; end
+
+  def update
```

Comment:
Ditto - unnecessary? Also `destroy` and `edit` - not sure when you'd want to edit one of these, unless you've got a different user story in mind.

Reply:
There's a `whois` field on this model, so you can edit to add whois data to the record.